### PR TITLE
Create test for clang-format

### PR DESCRIPTION
**Pull request by davetcoleman**
As discussed in https://github.com/ros-planning/moveit/issues/242

This is ready to merge - will not change any CI builds until the appropriate config flag is set

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,13 +53,18 @@ script:
...
-More configurations as seen in [industrial_ci](https://github.com/ros-industrial/industrial_ci) can be added, in the future.
+More configurations as seen in [industrial_ci](https://github.com/ros-industrial/industrial_ci) can be added in the future.
 
 ## Removed Configuration
 
 - ROS_REPOSITORY\_PATH: (UNSUPPORTED) replaced by ROS\_REPO
 
+## Clang-Format
+
+A new test is available that checks if the code is properly formatted as specified in the clang-format file found [here](https://github.com/davetcoleman/roscpp_code_format). Use ``TEST=clang-format`` to enable this test.
```
Comment (rhaschke) [inline README.md]: I think the format definition file should be part of the MoveIt repos too, probably moveit_ci.

File: travis.sh [hunk 2]
```diff
@@ -20,6 +20,12 @@ echo "Testing branch $TRAVIS_BRANCH of $REPOSITORY_NAME on $ROS_DISTRO"
 # Helper functions
 source ${CI_SOURCE_PATH}/$CI_PARENT_DIR/util.sh
 
+# Split for different tests
+if [ "${TEST}" == "clang-format" ]; then
```
Comment (rhaschke) [inline travis.sh]: You may want to support other tests in future. Rewrite this as:
```
for t in $TEST; do
  case "$t" in
     "clang-format") ... ;;
  esac
done
```

File: check_clang_format.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+echo "Running clang-format"
+find . -name '*.h' -or -name '*.hpp' -or -name '*.cpp' | xargs clang-format-3.6 -i -style=file
+
+echo "Showing results"
+travis_run git --no-pager diff
+
+# Make sure no changes have occured in repo
+if ! git diff-index --quiet HEAD --; then
+    # changes
+    echo "clang-format test failed: uncommitted changes detected in repo after running clang-format"
```
Comment (rhaschke) [inline check_clang_format.sh]: Consider an error message like: "clang-format test failed: changes required to comply to formatting rules"

File: README.md (hunk 1 above)
Comment (davetcoleman) [inline README.md]: good idea

**Comment by davetcoleman** [comment]
Good feedback - I've made your suggested fixes

**Pull request merged**